packages/react-core/src/components/TextInputGroup/__tests__/TextInputGroupMain.test.tsx

```diff
@@ -0,0 +1,278 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+
+import { TextInputGroupMain } from '../TextInputGroupMain';
+import { TextInputGroupContext } from '../TextInputGroup';
+import SearchIcon from '@patternfly/react-icons/dist/esm/icons/search-icon';
+
+describe('TextInputGroupMain', () => {
+  it('renders children', () => {
+    render(<TextInputGroupMain>Test</TextInputGroupMain>);
+
+    expect(screen.getByText('Test')).toBeVisible();
+  });
+
+  it('renders with class pf-c-text-input-group__main', () => {
+    render(<TextInputGroupMain>Test</TextInputGroupMain>);
+
+    const inputGroupMain = screen.getByText('Test');
+
+    expect(inputGroupMain).toHaveClass('pf-c-text-input-group__main');
+  });
+
+  it('does not render with class pf-m-icon when an icon prop is not passed', () => {
+    render(<TextInputGroupMain>Test</TextInputGroupMain>);
+
+    const inputGroupMain = screen.getByText('Test');
+
+    expect(inputGroupMain).not.toHaveClass('pf-m-icon');
+  });
+
+  it('renders with class pf-m-icon when an icon prop is passed', () => {
+    render(<TextInputGroupMain icon="icon">Test</TextInputGroupMain>);
+
+    const inputGroupMain = screen.getByText('Test');
+
+    expect(inputGroupMain).toHaveClass('pf-m-icon');
+  });
+
+  it('renders with custom class names provided via prop', () => {
+    render(<TextInputGroupMain className="custom-class">Test</TextInputGroupMain>);
+
+    const inputGroupMain = screen.getByText('Test');
+
+    expect(inputGroupMain).toHaveClass('custom-class');
+  });
+
+  it('renders with class pf-c-text-input-group__text on the inputs parent', () => {
+    render(<TextInputGroupMain>Test</TextInputGroupMain>);
+
+    const inputGroupText = screen.getByRole('textbox').parentNode;
+
+    expect(inputGroupText).toHaveClass('pf-c-text-input-group__text');
+  });
+
+  it('renders the input with class pf-c-text-input-group__text-input', () => {
+    render(<TextInputGroupMain>Test</TextInputGroupMain>);
+
+    const input = screen.getByRole('textbox');
+
+    expect(input).toHaveClass('pf-c-text-input-group__text-input');
+  });
+
+  it('renders given input icon props', () => {
```

Comment:
I think this is more-so "renders the icon" (or something like this) rather than this which reads to me essentially, "the text input group still renders when passing an icon prop". Maybe also mention that it is expected to be hidden.